scripts/train.py

```
@@ -0,0 +1,160 @@
+    elif color_type == 'E':
+        return COLOR_SEQ % (30 + COLORS['ERROR']) + string + RESET_SEQ
+    elif color_type == 'D':
+        return COLOR_SEQ % (30 + COLORS['DEBUG']) + string + RESET_SEQ
+
+def train_step_log(color_type: str, message: str):
+    warning_str = str_color(color_type, message)
+    logging.warning(warning_str)
+    warnings.warn(warning_str, UserWarning)
+
+class CustomRelation(RelationModel):
+    """ Fake relations
+    two nodes and rel
+    """
+    def __init__(self, nb_features: int, nb_labels: int, algebra: Algebra) -> None:
+        self.linear = torch.nn.Linear(2 * nb_features + nb_labels, algebra.flatdim)
+
+    @property
+    def parameters(self) -> Callable[[], Iterable[Any]]:
+        return self.linear.parameters
+
+    def __call__(self, x: Tsor, y: Tsor, l: Tsor) -> Tsor:
+        """ Compute x R y """
+        return self.linear(torch.cat((x, y, l), -1))
+
+class CustomScore(ScoringModel):
+    """ Must be defined. depends on labebra and 
+    in the scope of definition of the project. """
+    def __init__(
+            self,
+            nb_features: int,
+            nb_scores: int,
+            algebra: Algebra) -> None:
+        self.linear = torch.nn.Linear(
+            2 * nb_features + algebra.flatdim, nb_scores + 1)
+        self.softmax = torch.nn.Softmax(dim=-1)
+
+    @property
+    def parameters(self) -> Callable[[], Iterable[Any]]:
+        return self.linear.parameters
+
+    def __call__(self, src: Tsor, dst: Tsor, rel: Tsor) -> Tsor:
+        """ Compute S(src, dst, rel) """
+        cat_input = torch.cat((src, dst, rel), -1)
+        return self.softmax(self.linear(cat_input))[..., :-1]
+
+def train(ds_name: str, ds_path: str):
+    """[summary]
+    """
+    if ds_name == 'wn18':
+        ds = Dataset(path=ds_path, ds_name='wn18', node_vec_dim=10)
+    elif ds_name == 'fb15':
+        ds = Dataset(path=ds_path, ds_name='fb15', node_vec_dim=10)
```

Should we raise an exception for unknwon datasets?